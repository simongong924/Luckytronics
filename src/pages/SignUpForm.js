import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import {ReCaptcha} from 'react-recaptcha-google'

class SignUpForm extends Component {
    constructor() {
        super();
        this.onLoadRecaptcha = this.onLoadRecaptcha.bind(this);
        this.verifyCallback = this.verifyCallback.bind(this);

        this.state = {
            email: '',
            password: '',
            name: '',
            company: '',
            hasAgreed: false,
            ishuman : false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    componentDidMount(){
      if (this.captchaDemo) {
        console.log("just started");
        this.captchaDemo.reset();
      }
    }

    onLoadRecaptcha(){
      if (this.captchaDemo) {
        this.captchaDemo.reset();
        
      }
    }

    verifyCallback(recaptchaToken){
      console.log(recaptchaToken," <- This is your token");
    }
    
    handleValidation(){
        let email = this.state.email;
        let password = this.state.password;
        let name = this.state.name;
        let company = this.state.company;
        let errors = {};
        let formIsValid = true;

        //Name
        if(!email){
           formIsValid = false;
           errors["email"] = "Cannot be empty";
        }

        if (!password) {
            formIsValid= false;
            errors["password"] = "Cannot be empty";
        }

        if (!company) {
            formIsValid= false;
            errors["company"] = "Cannot be empty";
        }

        if (!name) {
            formIsValid= false;
            errors["name"] = "Cannot be empty";
        }
       this.setState({errors: errors});
       return formIsValid;
   }

    handleChange(e) {
        let target = e.target;
        let value = target.type === 'checkbox' ? target.checked : target.value;
        let name = target.name;
        let company = target.company;
        this.setState({
          [name]: value
        });
    }

    handleSubmit(e) {
        e.preventDefault();
        if (this.state.ishuman) {
        if (this.handleValidation()) {
          console.log(`Form submitted:`);

          const newTodo = {
              user_email: this.state.email,
              user_password: this.state.password,
              user_name: this.state.name,
              user_company: this.state.comapny,
              user_role: 'user',
              user_hasAgreed: this.state.hasAgreed
          }

          axios.post('http://localhost:5000/users/add', newTodo)
          .then(res => console.log(res.data));

          this.setState({
            user_email: '',
            user_password: '',
            user_name: '',
            user_company: '',
            user_role: '',
            user_hasAgreed: false
          })
          console.log('The form was submitted with the following data:');
          console.log(this.state);
          alert("User created successfully!");
        }} else{
            alert("Please Fill in all fields and verify captcha");
        }
    }

    render() {
        return (
        <div className="FormCenter">
            <h2> Register
            </h2>
            <form onSubmit={this.handleSubmit} className="FormFields">
              <div className="FormField">
                <label className="FormField__Label" htmlFor="name">Full Name</label>
                <input type="text" id="name" className="FormField__Input" placeholder="Enter your full name" name="name" value={this.state.name} onChange={this.handleChange} />
              </div>
              <div className="FormField">
                <label className="FormField__Label" htmlFor="password">Password</label>
                <input type="password" id="password" className="FormField__Input" placeholder="Enter your password" name="password" value={this.state.password} onChange={this.handleChange} />
              </div>
              <div className="FormField">
                <label className="FormField__Label" htmlFor="email">E-Mail Address</label>
                <input type="email" id="email" className="FormField__Input" placeholder="Enter your email" name="email" value={this.state.email} onChange={this.handleChange} />
              </div>
              <div className="FormField">
                <label className="FormField__Label" htmlFor="company">Company Name</label>
                <input type="company" id="company" className="FormField__Input" placeholder="Enter your comapny name" name="company" value={this.state.company} onChange={this.handleChange} />
              </div>
              <div className="FormField">
                  <button className="FormField__Button mr-20">Sign Up</button> 
              </div>
            </form>
            <ReCaptcha 
              sitekey="6Ld_Y58UAAAAANxW9B7B46ZthRasoC1Fqo3LdtcG"
              render="explicit"
              size="normal"
              onloadCallback={this.onLoadRecaptcha}
              verifyCallback={this.verifyCallback}
            />  
          </div>
        );
    }
}

export default SignUpForm;
