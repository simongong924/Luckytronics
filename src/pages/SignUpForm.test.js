import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-extended';
import SignUpForm from './SignUpForm';

//import { ENGINE_METHOD_STORE } from 'constants';
configure({ adapter: new Adapter() });

describe('<SignUpForm />',() => {
    let wrapper;
/*
it('has 2 inputs', () => {
const editor = shallow(<SignInForm />);
expect(editor.find('input').length).toEqual(2);
});
*/
it("renders without error",() =>{
const SignUpComponent = renderer.create(
<SignUpForm />).toJSON();

    expect(SignUpComponent).toMatchSnapshot();
});
it('check if default inputs are null/false on render',()=> {
    const props = {
        name : null,
        password : null,
        email : null,
        hasAgreed: false


    },
    SignUpComponent = mount(<SignUpForm {...props}/>);
    expect((SignUpComponent).prop('name')).toEqual(null);
    expect((SignUpComponent).prop('email')).toEqual(null);
    expect((SignUpComponent).prop('password')).toEqual(null);
    expect((SignUpComponent).prop('hasAgreed')).toBeFalsy();
});



it('check type of name',()=>{
    const john ='John Tan';
    const props ={
        name: 'John Tan'
    },

    SignUpComponent = mount(<SignUpForm {...props}/>);
    expect(SignUpComponent.prop('name')).toEqual(john);
    expect (SignUpComponent.prop('name')).toBeString();
});

it('submit empty form, receive error',()=>{
    wrapper = mount(<SignUpForm/>);
    wrapper.update();
    const form = wrapper.find('form');
    expect(form.simulate('submit').handleValidation).toBeFalsy();
});
/*
it('check submit error', ()=>{
    const props ={
        name:'John'

    },
    SignUpComponent = mount(<SignUpForm{...props}/>);
    submitform = SignUpComponent.find('Sign Up');
    submitform.simulate('click');
    expect(SignUpComponent.to)


}*/
});



/*
it('render correctly text component', () => {  
    const TextInputComponent = renderer.create(<SignUpForm />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});
*/