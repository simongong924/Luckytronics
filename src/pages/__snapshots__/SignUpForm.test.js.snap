// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders without error 1`] = `
<div
  className="FormCenter"
>
  <h2>
     Register
  </h2>
  <form
    className="FormFields"
    onSubmit={[Function]}
  >
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="name"
      >
        Full Name
      </label>
      <input
        className="FormField__Input"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Enter your full name"
        type="text"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="FormField__Input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="email"
      >
        E-Mail Address
      </label>
      <input
        className="FormField__Input"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="email"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <label
        className="FormField__Label"
        htmlFor="company"
      >
        Company Name
      </label>
      <input
        className="FormField__Input"
        id="company"
        name="company"
        onChange={[Function]}
        placeholder="Enter your comapny name"
        type="company"
        value=""
      />
    </div>
    <div
      className="FormField"
    >
      <button
        className="FormField__Button mr-20"
      >
        Sign Up
      </button>
    </div>
  </form>
</div>
`;
